Register binary attribute in collection.xconf.xsd

joewiz opened the pull request:
### Description:

Closes #5432 by adding schema definitions for the `field` element's `binary` attribute in the `collection.xconf.xsd` schema. 

### Reference: 

#5432

### Type of tests:

n/a

joewiz commented:
SonarCloud's failure:

> Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.20.0:site (default-site) on project exist: Failed to render site: Error generating maven-surefire-report-plugin:3.5.0:report report: Cannot invoke "Object.toString()" because "value" is null -> [Help 1]

The windows-latest checks haven't failed yet; they're still going, 30 minutes after submitting the PR...

dizzzz commented:
it is a recurring issue.... I guess a plugin update is needed...

duncdrum commented:
@joewiz did you mean issue to link to issue 5342 in the op, I corrected that as I don't see how that would be related

joewiz commented:
@duncdrum Thanks, yes! 

@dizzzz The `windows-latest` checks finished... after 40 minutes - wow!

adamretter reviewed (changes requested):
In principle this is fine, but I think perhaps we should just restrict the choices to "yes" and "no" so that it is consistent with the other elements in the schema.

line-o reviewed (commented):
Thanks @joewiz for taking this on. I second @adamretter's comment that the options should be limited to `yes` and `no` for consistency.
I am writing this knowing that collection configurations that include this option do set it to `true`. So we need to check first if `yes` will be recognised by the implementation and to change the documentation accordingly.

joewiz commented:
Thanks, all. I've removed `true` & `false` from the schema-allowed options and updated my PR branch. 

In [the issue](https://github.com/eXist-db/exist/issues/5432#issuecomment-2317738084) I linked to the implementation and its tests, which still respectively support and use `true` & `false`. I think those fixes should come in a separate PR. If that PR removes these values, it would constitute a breaking change, but of course the PR could also deprecate them, etc. 

(I think this PR still counts as a bugfix, since the old schema complained about the very presence of the `@binary` attribute.)
